vault/cors.go

```diff
@@ -0,0 +1,112 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var responseHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
+	"Access-Control-Max-Age":           "1800",
+	"Access-Control-Allow-Credentials": "true",
+	"Vary": "Origin",
+}
+
+var allowedMethods = []string{
+	http.MethodDelete,
+	http.MethodGet,
+	http.MethodOptions,
+	http.MethodPost,
+	http.MethodPut,
+}
+
+type CORSConfig struct {
```

Comment:
There needs to be a RWMutex guarding calls, because the moment someone live-toggles CORS config at runtime this races.